tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,787 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+                                , "List.isEmpty runs in constant time."
+                                ]
+                            , under = "List.length l /= 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (not << List.isEmpty) l
+"""
+                        ]
+        , test "should replace 0 /= List.length l with (not << List.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= List.length l
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << List.isEmpty)"
+                            , details =
+                                [ "Whereas List.length takes as long to run as the number of elements in the List,"
+                                , "List.isEmpty runs in constant time."
+                                ]
+                            , under = "0 /= List.length l"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (not << List.isEmpty) l
+"""
+                        ]
+        , test "should replace (l |> List.length) /= 0 with (not << List.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = (l |> List.length) /= 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << List.isEmpty)"
+                            , details =
+                                [ "Whereas List.length takes as long to run as the number of elements in the List,"
+                                , "List.isEmpty runs in constant time."
+                                ]
+                            , under = "(l |> List.length) /= 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (l |> (not << List.isEmpty))
+"""
+                        ]
+        , test "should replace 0 /= (l |> List.length) with (not << List.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= (l |> List.length)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << List.isEmpty)"
+                            , details =
+                                [ "Whereas List.length takes as long to run as the number of elements in the List,"
+                                , "List.isEmpty runs in constant time."
```

Comment:
Every element in the details list is a paragraph. Here, I think it's better for the two sentences to be in one paragraph (so a list with only a single string containing the whole message).

Reply:
roger! Was wondering how that all worked. Will replace with one line.